lib/puppet/type/cisco_vrf.rb

```diff
@@ -78,4 +79,19 @@ def self.title_patterns
     desc 'Shutdown state of the VRF.'
     newvalues(:true, :false)
   end # property shutdown
+
+  newproperty(:vni) do
+    desc "Specify virtual network identifier. Valid values are
+          integers between 4096 and 16777215, or keyword
+          'default' to disable this property"
+    munge do |value|
+      value = :default if value == 'default'
+      unless value == :default
+        value = value.to_i
+        fail 'vni value should be between 4096 and 16777215' unless
```

Comment:
In general I don't think we want to range check like this.  We should simply allow the device to reject invalid ranges.